phd-tests/framework/src/test_vm/mod.rs

```diff
@@ -627,6 +670,15 @@ impl TestVm {
     pub fn guest_os_has_read_only_fs(&self) -> bool {
         self.guest_os.read_only_fs()
     }
+
+    /// Generates a path to a file into which the VM's serial console adapter
+    /// can log serial console output.
+    fn serial_log_file_path(&self) -> Utf8PathBuf {
+        let filename = format!("{}.serial.log", self.spec.vm_name);
+        let mut path = self.data_dir.clone();
+        path.push(filename);
+        path
+    }
```

Comment:
as a follow-up, it might be nice if the PHD framework would go back and print the serial log output when a test fails, so that the output from failing tests ends up in the main log automatically?

i was also thinking about possibly changing PHD to do this for `tracing` logs by default, a la `libtest`'s output capturing...

Reply:
Printing the serial console output after the fact seems tricky in PHD's current architecture (though I might just be missing something). It's the runner that notices that test cases have failed (by observing an error returned from a test case or a panic caught while executing a test), so it would need to be the one to start dumping the consoles of any active VMs, but the runner has no idea what VMs the test owns. I think we'd need more control-plane-ish `Framework` that keeps track of the VMs it creates so that it can run them down on failure to do this elegantly.

> i was also thinking about possibly changing PHD to do this for tracing logs by default, a la libtest's output capturing...

I kind of like being able to follow the progress of the run in the terminal, but if there's a `--nocapture` equivalent I can always just add that to my PHD wrapper script.